plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/markerresolution/SuppressWarningFix.java

```diff
@@ -0,0 +1,88 @@
+package de.cognicrypt.staticanalyzer.markerresolution;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.TransformerException;
+
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.ui.IMarkerResolution;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.xml.sax.SAXException;
+
+import de.cognicrypt.core.Constants;
+import de.cognicrypt.staticanalyzer.Activator;
+import de.cognicrypt.utils.XMLParser;
+
+/**
+ * This class writes the suppress warning information in a XML file and deletes the marker on the UI
+ * @author André Sonntag
+ */
+public class SuppressWarningFix implements IMarkerResolution {
+
+	private String label;
+	private XMLParser xmlParser;
+
+	public SuppressWarningFix(String label) {
+		super();
+		this.label = label;
+	}
+
+	@Override
+	public String getLabel() {
+		return label;
+	}
+
+	@Override
+	public void run(IMarker marker) {
+
+		File warningsFile = new File(
+				marker.getResource().getProject().getLocation().toOSString() + "\\SuppressWarnings.xml");

```

Comment:
This would break on Unix machines. Use file separator constants from core.utils.Constants instead.